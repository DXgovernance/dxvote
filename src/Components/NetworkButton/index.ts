export { default as NetworkButton } from './NetworkButton';
