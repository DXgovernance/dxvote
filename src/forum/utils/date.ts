import moment from 'moment';

export const formatTime = date => {
  const d = new Date(date);
  return moment(d).format('MMM dd');
};

