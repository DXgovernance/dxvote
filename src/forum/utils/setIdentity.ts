import Identities from 'orbit-db-identity-provider';
import { getResolver } from '@ceramicnetwork/3id-did-resolver';

// Connects the logged in Ceramic user to OrbitDB
export async function setIdentity({ orbitdb, registry }, { ceramic, threeId }) {
  if (orbitdb.identity.type !== 'did') {
    Identities.DIDIdentityProvider.setDIDResolver(getResolver(ceramic));
    const identity = await Identities.createIdentity({
      type: 'DID',
      didProvider: threeId.getDidProvider(),
    });
    orbitdb.identity = identity;
    registry.setIdentity(identity);
    console.log('OrbitDB: DID identity set!', identity);
  }
}
