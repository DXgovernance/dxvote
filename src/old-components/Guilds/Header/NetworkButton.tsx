import { useState } from 'react';
import { useWeb3React } from '@web3-react/core';

import { getChains, isChainIdSupported } from 'provider/connectors';
import { Button, ButtonIcon, IconButton } from '../common/Button';
import NetworkModal from '../Web3Modals/NetworkModal';
import arbitrumIcon from '../../../assets/images/arbitrum.png';
import ethereumIcon from '../../../assets/images/ether.svg';
import gnosisIcon from '../../../assets/images/gnosis-icon-green.svg';

export const iconsByChain = {
  1: ethereumIcon,
  4: ethereumIcon,
  100: gnosisIcon,
  42161: arbitrumIcon,
  421611: arbitrumIcon,
  1337: ethereumIcon,
};

const NetworkButton = () => {
  const [isNetworkModalOpen, setIsNetworkModalOpen] = useState(false);

  const { chainId, error } = useWeb3React();
  const chainName =
    getChains().find(chain => chain.id === chainId)?.displayName || null;

  const toggleNetworkModal = () => {
    setIsNetworkModalOpen(!isNetworkModalOpen);
  };

  function getNetworkStatus() {
    if ((chainId && !isChainIdSupported(chainId)) || error) {
      return <Button onClick={toggleNetworkModal}>Unsupported Network</Button>;
    } else if (chainId) {
      return (
        <IconButton onClick={toggleNetworkModal} iconLeft>
          <ButtonIcon src={iconsByChain[chainId]} alt={'Icon'} />
          {chainName}
        </IconButton>
      );
    } else {
      return (
        <Button onClick={toggleNetworkModal} active={true}>
          Not Connected
        </Button>
      );
    }
  }

  return (
    <>
      {getNetworkStatus()}
      <NetworkModal isOpen={isNetworkModalOpen} onClose={toggleNetworkModal} />
    </>
  );
};

export default NetworkButton;
