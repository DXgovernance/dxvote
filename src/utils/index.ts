export * from './address';
export * from './bignumber';
export * from './cache';
export * from './constants';
export * from './date';
export * from './enums';
export * from './helperHooks';
export * from './helpers';
export * from './number';
export * from './permissions';
export * from './proposals';
export * from './scheme';
export * from './transaction';
export * from './encodingCalls';
export * from './signature';
export * from './array';
export * from './votes';
