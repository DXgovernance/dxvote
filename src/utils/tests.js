import { ThemeProvider } from 'styled-components';
import { render as rtlRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { GuildsDarkTheme } from 'Components/theme';

export const renderWithRoutes = (ui, options = {}) => {
  //wrap with MemoryRouter
  return rtlRender(ui, { wrapper: MemoryRouter, ...options });
};

// we are mokcing testing-library/render with our own render
export const render = (children, options = {}) => {
  //wrap with Theme
  return renderWithRoutes(
    <ThemeProvider theme={GuildsDarkTheme}> {children} </ThemeProvider>,
    options
  );
};
