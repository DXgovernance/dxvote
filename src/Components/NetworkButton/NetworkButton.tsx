import { useState } from 'react';
import { useWeb3React } from '@web3-react/core';
import { useTranslation } from 'react-i18next';

import { isChainIdSupported } from 'provider/connectors';
import { getNetworkById, getChainIcon } from 'utils';
import {
  Button,
  ButtonIcon,
  IconButton,
} from 'old-components/Guilds/common/Button';
import NetworkModal from 'old-components/Guilds/Web3Modals/NetworkModal';

const NetworkButton = () => {
  const { t } = useTranslation();
  const [isNetworkModalOpen, setIsNetworkModalOpen] = useState(false);

  const { chainId, error } = useWeb3React();
  const chainName = getNetworkById(chainId)?.displayName || null;

  const toggleNetworkModal = () => {
    setIsNetworkModalOpen(!isNetworkModalOpen);
  };

  function getNetworkStatus() {
    if ((chainId && !isChainIdSupported(chainId)) || error) {
      return (
        <Button onClick={toggleNetworkModal}>{t('unsupportedNetwork')}</Button>
      );
    } else if (chainId) {
      return (
        <IconButton onClick={toggleNetworkModal} iconLeft>
          <ButtonIcon src={getChainIcon(chainId)} alt={'Icon'} />
          {chainName}
        </IconButton>
      );
    } else {
      return <Button onClick={toggleNetworkModal}>{t('notConnected')}</Button>;
    }
  }

  return (
    <>
      {getNetworkStatus()}
      <NetworkModal isOpen={isNetworkModalOpen} onClose={toggleNetworkModal} />
    </>
  );
};

export default NetworkButton;
