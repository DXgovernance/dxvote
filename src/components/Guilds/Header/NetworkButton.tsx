import styled from 'styled-components';
import { observer } from 'mobx-react';
import { isChainIdSupported } from 'provider/connectors';
import NetworkModal from 'components/NetworkModal';
import { useContext } from '../../../contexts';
import { toCamelCaseString } from '../../../utils';
import { Button, IconButton } from '../common/Button';

import arbitrumIcon from '../../../assets/images/arbitrum.png';
import ethereumIcon from '../../../assets/images/ethereum.svg';
import xdaiIcon from '../../../assets/images/xdai.svg';

const iconsByChain = {
  1: ethereumIcon,
  4: ethereumIcon,
  100: xdaiIcon,
  42161: arbitrumIcon,
  421611: arbitrumIcon,
  1337: ethereumIcon,
};

const ButtonIcon = styled.img`
  height: 1.1rem;
  width: 1.1rem;
`;

const NetworkButton = observer(() => {
  const {
    context: { modalStore, providerStore, configStore },
  } = useContext();

  const { chainId, error } = providerStore.getActiveWeb3React();

  const toggleNetworkModal = () => {
    modalStore.toggleNetworkModal();
  };

  function getNetworkStatus() {
    if ((chainId && !isChainIdSupported(chainId)) || error) {
      return <Button onClick={toggleNetworkModal}>Unsupported Network</Button>;
    } else if (chainId) {
      return (
        <IconButton onClick={toggleNetworkModal} iconLeft>
          <ButtonIcon src={iconsByChain[chainId]} alt={'Icon'} />
          {toCamelCaseString(configStore.getActiveChainName())}
        </IconButton>
      );
    } else {
      return (
        <Button onClick={toggleNetworkModal} active={true}>
          Not Connected
        </Button>
      );
    }
  }

  return (
    <>
      {getNetworkStatus()}
      <NetworkModal />
    </>
  );
});

export default NetworkButton;
