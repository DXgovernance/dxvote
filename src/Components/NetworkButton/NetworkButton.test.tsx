import NetworkButton from './NetworkButton';
import { render } from 'utils/tests';
import { useWeb3React } from '@web3-react/core';

jest.mock('contexts/index', () => jest.fn());
jest.mock('provider/providerHooks', () => {
  return {
    useRpcUrls: () => {
      return {
        '1': 'https://mainnet.infura.io/v3/dummy',
      };
    },
  };
});

jest.mock('@web3-react/core', () => ({
  useWeb3React: jest.fn(),
}));

const mockedUseWeb3React = useWeb3React as jest.Mock;

describe('NetworkButton', () => {
  it('Should match snapshot and display not connected status', () => {
    mockedUseWeb3React.mockImplementation(() => ({
      active: false,
      chainId: null,
    }));
    const { container, getByText } = render(<NetworkButton />);
    expect(container).toMatchSnapshot();
    expect(getByText('notConnected')).toBeInTheDocument();
  });
  it('Should match snapshot and display network', () => {
    mockedUseWeb3React.mockImplementation(() => ({
      active: true,
      chainId: 1,
    }));
    const { container, getByText } = render(<NetworkButton />);
    expect(container).toMatchSnapshot();
    expect(getByText('Ethereum Mainnet')).toBeInTheDocument();
  });
});
