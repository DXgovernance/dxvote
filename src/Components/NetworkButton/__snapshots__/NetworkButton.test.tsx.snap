// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkButton Should match snapshot and display network 1`] = `
<div>
   
  <button
    class="sc-bdfBQB sc-gsTEea cnzpap hqMJhH"
  >
    <img
      alt="Icon"
      class="sc-dlfnuX hZGubn"
      src="ethereum.svg"
    />
    Ethereum Mainnet
  </button>
   
</div>
`;

exports[`NetworkButton Should match snapshot and display not connected status 1`] = `
<div>
   
  <button
    class="sc-bdfBQB cnzpap"
  >
    guilds.notConnected
  </button>
   
</div>
`;
